src/components/app-switcher/app-switcher.js

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Renders an application drawers that can be used to switch programs
+ * @class AppSwitcher
+ */
+
+export default class AppSwitcher extends React.Component {
+
+  static propTypes = {
+        /**
+         * The json blob for the application links
+         *
+         * @property applicationJson
+         * @type {String}
+         * @default ''
+         */
+    applicationJson: PropTypes.string.isRequired,
+
+        /**
+         * The title of the menu item
+         *
+         * @property menuTitle
+         * @type {String}
+         * @default ''
+         */
+    menuTitle: PropTypes.string
+
+  }
+
+    /**
+     * Return the classes for the styles
+     *
+     * @method styles
+     * @return Hash
+     */
+
+  get styles() {
+    const prefix = 'react-switcher-';
+    return {
+      appItem: prefix + "app-item",
+      menu: classNames((this.state.active ? prefix + "menu-active" : ""), prefix + "menu-title"),
+      sectionTitle: prefix + "section-title",
+      drawer: prefix + "drawer",
+      arrowUp: prefix + "arrow-up",
+      symbol: prefix + "symbol"
+    };
+  }
+
+    /**
+     * Constructor method setting properties
+     *
+     * @method constructor
+     * @return {Void}
+     */
+
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
```

Comment:
I removed the application json from inside the state. It only needs to be a prop anyways. As for 'active', I can move that to flux, but it is only currently referenced internally by its own component.